src/CodeGen_ARM.cpp

```diff
@@ -770,6 +840,22 @@ void CodeGen_ARM::init_module() {
     }
 }
 
+void CodeGen_ARM::compile_func(const LoweredFunc &f,
+                               const string &simple_name,
+                               const string &extern_name) {
+    CodeGen_Posix::begin_func(f.linkage, simple_name, extern_name, f.args);
+
+    Stmt body = f.body;
+
+    if (target.os != Target::IOS && target.os != Target::OSX) {
+        body = SubstituteInStridedLoads().mutate(body);
```

Comment:
A comment explaining why we don't perform this substitution for IOS or OSX would be helpful here